src/hsm.cpp

```diff
@@ -63,12 +64,26 @@ HsmEngine::HsmEngine(const std::string &id,
 
 HsmEngine::~HsmEngine() { _ENGINE_finish(_engine.get()); }
 
+std::string HsmEngine::_constructPkcs11URI(const std::string &keyLabel,
```

Comment:
I'd prefer an API that features std::optional or an overload with/without key label. Having the empty string special is somewhat ugly.